web/src/buddy_list.js

```diff
@@ -55,16 +87,259 @@ class BuddyListConf {
 
 export class BuddyList extends BuddyListConf {
     all_user_ids = [];
+    users_matching_view_ids = [];
+    other_user_ids = [];
+    tooltips_initialized = false;
+    users_matching_view_is_collapsed = false;
+    other_users_is_collapsed = false;
+
+    initialize_tooltips() {
+        $("#right-sidebar").on("mouseenter", ".buddy-list-heading", (e) => {
+            e.stopPropagation();
+            const $elem = $(e.currentTarget);
+            let placement = "left";
+            if (window.innerWidth < media_breakpoints_num.md) {
+                // On small devices display tooltips based on available space.
+                // This will default to "bottom" placement for this tooltip.
+                placement = "auto";
+            }
+            const get_total_subscriber_count = this.total_subscriber_count;
+            tippy($elem[0], {
+                // Because the buddy list subheadings are potential click targets for purposes
+                // having nothing to do with the subscriber count (collapsing/expanding), we
+                // delay showing the tooltip until the user lingers a bit longer.
+                delay: INTERACTIVE_HOVER_DELAY,
+                // Don't show tooltip on touch devices (99% mobile) since touch pressing on users in the left or right
+                // sidebar leads to narrow being changed and the sidebar is hidden. So, there is no user displayed
+                // to show tooltip for. It is safe to show the tooltip on long press but it not worth
+                // the inconvenience of having a tooltip hanging around on a small mobile screen if anything going wrong.
```

Comment:
These comment lines should be wrapped to more like 80 characters. We allow longer lines for URLs and  places where it's more readable that way, but this is definitely not more readable written this way.

Reply:
Fixed!